src/plugins/toml/README.md

```diff
@@ -0,0 +1,210 @@
+Arrays are recognized by the `array` metakey. On writing, the plugin will detect arrays automatically and set the appropriate metakey if it is missing.
+TODO: Example
+
+# Order
+
+The plugin preserves the file order by the usage of the metakey `order`. When reading a file, the order metakey will be set according to the order as read in the file.
+If new keys are added, eg. via `kdb set`, the order of the set key will be set to the next-to-highest order value present in the existing key set.
+However, the order is only relevant between elements with the same TOML-parent. For example keys of a simple table are only sorted with respecet to each other, not with any keys outside that table. If that table has it's order changed and moves to another position in the file, so will it's subkeys.
+
+There is an additional limitation on ordering for prevention of messing up resulting TOML-file semantics on writing.
+When sorting elements under the same TOML-parent, tables (simple and array) will always be sorted after non-table elements, regardless of their order.
+With this limitation, we prevent that a newly set key, that is not part of a certain table array/simple table, would be placed after the table declaration, making it a member of that table on a subsequent read.
+
+```
+# Mount TOML file
+sudo kdb mount test_strings.toml user/tests/storage toml type
+
+# Create three keys in reverse alphabetical order under the subkey common
+# Additionally, create one key not in the common subkey space
+kdb set 'user/tests/storage/common/c' '0'
+kdb set 'user/tests/storage/common/b' '1'
+kdb set 'user/tests/storage/common/a' '2'
+kdb set 'user/tests/storage/d' '3'
+
+# Print the content of the resulting TOML file
+# The keys are ordered as they were set
+cat `kdb file user/tests/storage`
+#> common.c = 2
+#> common.b = 1
+#> common.a = 0
+#> d = 3
+
+# Create a simple table for the three keys under `common`
+kdb meta-set 'user/tests/storage/common' 'tomltype' 'simpletable'
+
+# Print the content of the resulting TOML file
+cat `kdb file user/tests/storage`
+#> d = 3
+#> [common]
+#> c = 0
+#> b = 1
+#> a = 2
+
+# Cleanup
+kdb rm -r user/tests/storage
+sudo kdb umount user/tests/storage
+```
+
+In this example, `d` and `common` have the same parent (the file root). This means, they need to be sorted with each other. `d` would be placed before `common` by it's order (since it was set before, and thus, has lesser order).
+Their order however never gets compared, since `common` is a simple table and `d` is not, so `d` will get sorted before the table regardless of order.
+
+TODO: Maybe explain better/more clear/shorter?
+
+# Limitations
+
+- Comments and newlines between the last array element and closing brackets are discarded.
+- Trailing commas in arrays and inline tables are discarded
+- Only spaces in front of comments are preserved.
+- Currently, Elektra's sparse arrays are not preserved on writing (they get a continuous array without index holes).
+
+# TODOs
```

Comment:
Write the README for the end-user of your plugin. Move important TODOs to the issue tracker. Small TODOs in the code are fine of course.